### PR TITLE
Feature/dev 844 recycle not ready nodes

### PR DESCRIPTION
**Pull request by MacLikorne**
The commented stuff are waiting for more testing in real situation.

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	}
+}
+
+func getKubeClient(cmd *cobra.Command, kubernetesConn string) (*kubernetes.Clientset,dynamic.Interface) {
+	var k8sClientSet *kubernetes.Clientset
+	var dynamicClient dynamic.Interface
+	var err error
+
+	switch kubernetesConn {
+	case "in":
```
Comment (deimosfr) [inline cmd/run.go]: in is useless here as there are only 2 possibilities that you already checked during the arg check

File: utils/go.mod [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module ithub.com/Qovery/do-k8s-replace-notready-nodes/utils
```
Comment (deimosfr) [inline utils/go.mod]: type -> github?

File: utils/k8s.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+* created more than five minutes ago.
+ */
+func checkNode(node v12.Node) FullNode{
+	var fullNode FullNode
+
+	for _, condition := range node.Status.Conditions {
+		if strings.EqualFold(fmt.Sprint(condition.Type), "Ready") &&
+			strings.EqualFold(fmt.Sprint(condition.Status), "Unknown") &&
+			!node.Spec.Unschedulable &&
+			node.CreationTimestamp.Add(time.Minute *5).Before(time.Now()) {
```
Comment (deimosfr) [inline utils/k8s.go]: can you please make this field configurable (with env vars for example). if we see 5 min is not enough/too much, we'll easily be able to update it

File: utils/k8s.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+
+	for event := range result.ResultChan() {
+		if event.Type == watch.Added || event.Type == watch.Modified {
+			node := getNodesInfosFromRuntineObject(dynClinent, "default", event.Object)
+			checkedNode := checkNode(node)
+			emptyNode := FullNode{}
+			if checkedNode != emptyNode {
+				//RecyleNode(DOclient, checkedNode)
```
Comment (deimosfr) [inline utils/k8s.go:140]: You shoud uncomment it ;)

File: utils/k8s.go [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+			}
+		}
+	}
+}
+
+func getNodesInfosFromRuntineObject(dynClient dynamic.Interface, nameSpace string, obj runtime.Object) v12.Node{
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+	if err != nil {
+		log.Errorf("Can't unstructure object: %s", err)
```
Comment (deimosfr) [inline utils/k8s.go]: Please give more context (ex: kind of object)

File: utils/k8s.go [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"strings"
+	"time"
+)
+
+type FullNode struct {
```
Comment (deimosfr) [inline utils/k8s.go]: FullNode means?...NodeInfos? If yes, may be renaming it to make it more clear is better

File: utils/workers.go [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+		go result(done, results)
+		createWorkerPool(workerCount, &jobParams, job, &results)
+		<- done
+
+		log.Debug(endMessage)
+}
+
+
+// TODO: delete when everything works
+func workerTestFunc(params ...interface{}) interface{} {
```
Comment (deimosfr) [inline utils/workers.go]: You should seprate your tests. Go convention is keeping the same name of the file and adding _test

File: utils/k8s.go (hunk 6 above)
Comment (MacLikorne) [reply utils/k8s.go]: done

File: utils/workers.go (hunk 7 above)
Comment (MacLikorne) [reply utils/workers.go]: it was about to be deleted

File: utils/k8s.go (hunk 5 above)
Comment (MacLikorne) [reply utils/k8s.go]: done

File: utils/k8s.go (hunk 4 above)
Comment (MacLikorne) [reply utils/k8s.go:140]: I'm scared, not 100% sure the filter get only stucknodes atm

File: utils/go.mod (hunk 2 above)
Comment (MacLikorne) [reply utils/go.mod]: corrected

File: utils/k8s.go (hunk 3 above)
Comment (MacLikorne) [reply utils/k8s.go]: done

File: cmd/run.go (hunk 1 above)
Comment (MacLikorne) [reply cmd/run.go]: it was legacy code, was afraid to delete it

File: utils/k8s.go (hunk 4 above)
Comment (deimosfr) [reply utils/k8s.go:140]: Ok, may be then add a log saying it should remove those nodes, what do you think?

Comment (MacLikorne) [reply utils/k8s.go:140]: sure

**Pull request merged**